src/chemenv/tools/converters.py

```diff
@@ -0,0 +1,387 @@
+        """
+        url = f"https://pubchem.ncbi.nlm.nih.gov/rest/pug/compound/name/{self.name}/property/IsomericSMILES/JSON"
+        
+        async with aiohttp.ClientSession() as session:
+            try:
+                async with session.get(url, timeout=self.timeout) as response:
+                    if response.status == 200:
+                        data = await response.json()
+                        return data["PropertyTable"]["Properties"][0]["IsomericSMILES"]
+                    logger.warning(f"PubChem API failed with status {response.status}")
+                    return None
+            except Exception as e:
+                logger.error(f"PubChem API error: {e}")
+                print(e)
+                return None
+
+    @backoff.on_exception(backoff.expo, (aiohttp.ClientError, asyncio.TimeoutError), max_time=10, logger=logger)
+    async def unknown(self) -> Optional[str]:
+        """
+        Queries the Unknown API to convert chemical name to SMILES.
+
+        Returns:
+            str: SMILES notation if successful, None otherwise
+        
+        Raises:
+            aiohttp.ClientError: On API connection errors
+            asyncio.TimeoutError: If request times out
+        """ 
+        url = f"http://46.4.119.202:8082/?name=%7B{self.name}%7D&what=smiles"
+        
+        async with aiohttp.ClientSession() as session:
+            try:
+                async with session.get(url, timeout=self.timeout) as response:
+                    if response.status == 200:
+                        message_content = await response.text()
+                        message_text = message_content.split('Message:')[1].strip()
+                        return message_text
+                    logger.warning(f"Unknown API failed with status {response.status}")
+                    return None
+            except Exception as e:
+                logger.error(f"Unknown API error: {e}")
+                return None
+    
+
+    async def get_smiles(self) -> Optional[str]:
+        """Query all APIs in parallel until a valid SMILES is found.
+        
+        Attempts to convert name to SMILES using multiple APIs concurrently,
+        returning the first successful result.
+        
+        Returns:
+            str: First valid SMILES notation found
+            
+        Raises:
+            ValueError: If no API returns a valid SMILES notation
+        """
+        tasks = [
+            self.opsin(),
+            self.cactus(),
+            self.pubchem(),
+            self.unknown(),
+        ]
+        
+        for result in asyncio.as_completed(tasks):
+            try:
+                smiles = await result
+                if smiles:
+                    return smiles
+            except Exception as e:
+                logger.error(f"Error in get_smiles: {e}")
+                continue
+        
+        raise ValueError(f"Could not find SMILES for {unquote(self.name)}")
+
+
+class Smiles2Name:
```

Comment:
This is mostly the same as for `Name2Smiles` and both could be merged, but I think it would be better to keep them simple